### PR TITLE
`ContractCreate` and `ContractCall` preTransactionHandler 

### LINKED ISSUE #4549: Implement ContractCallHandler `preHandle()` method

### LINKED ISSUE #4550: Implement ContractCreateHandler `preHandle()` method

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #4505 
Fixes #4549 
- Adds `ContractCreate` and `ContractCall` preTransactionHandler

### CODE REVIEW

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/DispatcherTest.java
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/DispatcherTest.java:294]: Is the newline needed by spotless?

File: hedera-node/hedera-smart-contract-service-impl/build.gradle.kts
Comment (hendrikebbers) [inline hedera-node/hedera-smart-contract-service-impl/build.gradle.kts:34]: What is this doing?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java
Comment (hendrikebbers) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java:56]: Objects.requireNotNull for txBody

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java
Comment (hendrikebbers) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java:60]: You do a cast to JKey without checking if it is an instance of

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java
Comment (hendrikebbers) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java:48]: I would define them as `protected final` or `private` and add getter/setter.

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java
Comment (hendrikebbers) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java:58]: Should it be possible to overwrite the mock instances in a child class? I would define them as protected final or private and add getter/setter.

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java
Comment (tinker-michaelj) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java:63]: Imo we can leave off this `!op.getAutoRenewAccountId().equals(MISSING_ACCOUNT_ID)` test; it's just a small optimization to save calling `addNonPayerKey()` when it's guaranteed to fail; but seems very arbitrary. 

(I.e. why don't we also check if `accountNum < 0`, or `shardNum != STATIC_PROPERTIES_HOLDER.shard()`, etc., etc...?)

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCreateHandlerParityTest.java
Comment (tinker-michaelj) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCreateHandlerParityTest.java:49]: Awesome. ❤️

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/DispatcherTest.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/dispatcher/DispatcherTest.java:294]: I think so.. Spotless did that change 🤔

**Review by qnswirlds — DISMISSED** [review]
LGTM overall!

File: hedera-node/hedera-smart-contract-service-impl/build.gradle.kts
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-smart-contract-service-impl/build.gradle.kts:34]: I think I added it for some dependency. Removed it, as it is not needed anymore

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java:60]: All the `HederaKey`s are currently of type `JKey`. This cast will be removed once we remove legacy JKey class

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java:48]: Added final

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractHandlerTestBase.java:58]: I think all the Handlers need them and don't need to overwrite. I could not add final as it requires to be added in constructor.

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java:63]: Agreed. Removed that check in `ContractCreate` preHandle

**Review by mhess-swl — APPROVED** [review]

**Comment by Neeharika-Sompalli** [comment]
Will merge this once States API PR is merged

**Review by qnswirlds — APPROVED** [review]

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/AdapterUtils.java
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/AdapterUtils.java:36]: Maybe we should put a few emojis or !'s to make this more prominent 😂 

```suggestion
// NOTE: This class is duplicated in more than one service module.
// !!!!!!!!!!🔥🔥🔥 It should be deleted once we find where to keep it. 🔥🔥🔥!!!!!!!!!!!
```

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/module-info.java
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/module-info.java:17]: Per @hendrikebbers suggestion on a different PR, I _think_ we can use `open module` to get rid of both `opens` lines:

```
open module com.hedera.node.app.service.contract.impl.test {
    requires com.hedera.node.app.service.contract;
    requires com.hedera.node.app.service.contract.impl;
    requires org.junit.jupiter.api;
    requires com.hedera.node.app.service.mono;
    requires com.hedera.node.app.service.mono.testFixtures;
    requires org.mockito;
    requires org.hamcrest;
    requires org.assertj.core;
    requires org.mockito.junit.jupiter;
    requires com.hedera.node.app.spi;
    requires hedera.services.hedera.node.hedera.app.spi.testFixtures;
}

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/module-info.java
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/module-info.java:17]: I guess it opens module to everything in addition to `org.junit.platform.commons` right ?

**Review by qnswirlds — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/module-info.java
Comment (mhess-swl) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/module-info.java:17]: Yes, but since it's a test module I'm not sure we care. Not a big deal either way though

**Pull request merged**
